### ISSUE TITLE
label_image segmentation issue for operator code 25 which is softmax

### ISSUE THREAD

**Issue by praveen-dn** [tensorflow/tensorflow#25314]
When using the standard label_image to test the mobilenet tflite file in the default P OS android NN.
we are facing the segmentation issue for the softmax operation with the default command as specified in the readme.md.

We are using latest Android NN on P Os, for Arm64-v8a series soc.

The following is the crash snippet:
```
01-24 15:28:12.701 I/ExecutionBuilder( 4787): ExecutionBuilder::startCompute (from plan, iteratively)
01-24 15:28:12.701 I/ExecutionBuilder( 4787): looking for next StepExecutor
01-24 15:28:12.702 I/ExecutionPlan( 4787): ExecutionPlan::next(0x7c00830460, 0x7bfe80c308): mNextStepIndex = 0
01-24 15:28:12.702 I/ExecutionBuilder( 4787): input[0] = POINTER(0x7bfcf4ac00)
01-24 15:28:12.702 I/ExecutionBuilder( 4787): output[0] = POINTER(0x7bfce00fc0)
01-24 15:28:12.707 I/CpuExecutor( 4787): CpuExecutor::run() with request({.inputs = [1]{{.hasNoValue = 0, .location = {.poolIndex = 0, .offset = 0, .length = 602112}, .dimensions = [4]{1, 224, 224, 3}}}, .outputs = [1]{{.hasNoValue = 0, .location = {.poolIndex = 1, .offset = 0, .length = 4004}, .dimensions = [2]{1, 1001}}}, .pools = [0]{}})
01-24 15:28:12.707 I/CpuExecutor( 4787): CpuExecutor::initializeRunTimeInfo
01-24 15:28:12.763 D/SDHMS:ComponentLevelSetter(22281): mWifiTotalUsage = 0, throughput = 0.0
01-24 15:28:12.928 I/CpuExecutor( 4787): Completed run normally
01-24 15:28:12.932 I/ExecutionBuilder( 4787): looking for next StepExecutor
01-24 15:28:12.932 I/ExecutionPlan( 4787): ExecutionPlan::next(0x7c00830460, 0x7bfe80c308): mNextStepIndex = 1
01-24 15:28:12.934 I/ExecutionBuilder( 4787): ExecutionBuilder::ExecutionBuilder
01-24 15:28:12.936 I/ExecutionBuilder( 4787): ExecutionBuilder::startCompute (from plan, iteratively)
01-24 15:28:12.936 I/ExecutionBuilder( 4787): looking for next StepExecutor
01-24 15:28:12.936 I/ExecutionPlan( 4787): ExecutionPlan::next(0x7c00830460, 0x7bfe80c308): mNextStepIndex = 0
01-24 15:28:12.936 I/ExecutionBuilder( 4787): input[0] = POINTER(0x7bfcf4ac00)
01-24 15:28:12.936 I/ExecutionBuilder( 4787): output[0] = POINTER(0x7bfce00fc0)
01-24 15:28:12.940 I/CpuExecutor( 4787): CpuExecutor::run() with request({.inputs = [1]{{.hasNoValue = 0, .location = {.poolIndex = 0, .offset = 0, .length = 602112}, .dimensions = [4]{1, 224, 224, 3}}}, .outputs = [1]{{.hasNoValue = 0, .location = {.poolIndex = 1, .offset = 0, .length = 4004}, .dimensions = [2]{1, 1001}}}, .pools = [0]{}})
01-24 15:28:12.941 I/CpuExecutor( 4787): CpuExecutor::initializeRunTimeInfo
01-24 15:28:13.295 I/UiThreadMonitor(20720): setAwake 103 5001
01-24 15:28:13.301 I/CpuExecutor( 4787): Completed run normally
01-24 15:28:13.305 I/ExecutionBuilder( 4787): looking for next StepExecutor
01-24 15:28:13.305 I/ExecutionPlan( 4787): ExecutionPlan::next(0x7c00830460, 0x7bfe80c308): mNextStepIndex = 1
01-24 15:28:13.306 I/ExecutionBuilder( 4787): ExecutionBuilder::ExecutionBuilder
01-24 15:28:13.307 I/ExecutionBuilder( 4787): ExecutionBuilder::startCompute (from plan, iteratively)
01-24 15:28:13.307 I/ExecutionBuilder( 4787): looking for next StepExecutor
01-24 15:28:13.307 I/ExecutionPlan( 4787): ExecutionPlan::next(0x7c00830460, 0x7bfe80c308): mNextStepIndex = 0
01-24 15:28:13.307 I/ExecutionBuilder( 4787): input[0] = POINTER(0x7bfcf4ac00)
01-24 15:28:13.307 I/ExecutionBuilder( 4787): output[0] = POINTER(0x7bfce00fc0)
01-24 15:28:13.308 I/CpuExecutor( 4787): CpuExecutor::run() with request({.inputs = [1]{{.hasNoValue = 0, .location = {.poolIndex = 0, .offset = 0, .length = 602112}, .dimensions = [4]{1, 224, 224, 3}}}, .outputs = [1]{{.hasNoValue = 0, .location = {.poolIndex = 1, .offset = 0, .length = 4004}, .dimensions = [2]{1, 1001}}}, .pools = [0]{}})
01-24 15:28:13.308 I/CpuExecutor( 4787): CpuExecutor::initializeRunTimeInfo
01-24 15:28:13.547 I/CpuExecutor( 4787): Completed run normally
01-24 15:28:13.551 I/ExecutionBuilder( 4787): looking for next StepExecutor
01-24 15:28:13.551 I/ExecutionPlan( 4787): ExecutionPlan::next(0x7c00830460, 0x7bfe80c308): mNextStepIndex = 1
--------- beginning of crash
01-24 15:28:13.554 F/libc    ( 4787): Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x3d38 in tid 4787 (label_image), pid 4787 (label_image)
01-24 15:28:13.636 I/crash_dump64( 5086): obtaining output fd from tombstoned, type: kDebuggerdTombstone
01-24 15:28:13.639 I//system/bin/tombstoned( 5616): received crash request for pid 4787
01-24 15:28:13.641 I/crash_dump64( 5086): performing dump of process 4787 (target tid = 4787)
01-24 15:28:13.644 F/DEBUG   ( 5086): *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
01-24 15:28:13.645 F/DEBUG   ( 5086): Build fingerprint: 'samsung/beyond2ltexx/beyond2:9/PPR1.180610.011/G975FXXE1ASAM:eng/test-keys'
01-24 15:28:13.645 F/DEBUG   ( 5086): Revision: '20'
01-24 15:28:13.645 F/DEBUG   ( 5086): ABI: 'arm64'
01-24 15:28:13.645 F/DEBUG   ( 5086): pid: 4787, tid: 4787, name: label_image  >>> ./label_image <<<
01-24 15:28:13.645 F/DEBUG   ( 5086): signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x3d38
01-24 15:28:13.646 F/DEBUG   ( 5086):     x0  0000005dc122ebf8  x1  0000000000000000  x2  0000000000000000  x3  0000007c00ef33c8
01-24 15:28:13.646 F/DEBUG   ( 5086):     x4  0000007fd73faad8  x5  0000000000000020  x6  0000007fd73faaa8  x7  0000000000000010
01-24 15:28:13.646 F/DEBUG   ( 5086):     x8  0000000000003d38  x9  385838b172084ce5  x10 00000000000012b3  x11 00000000000012b3
01-24 15:28:13.646 F/DEBUG   ( 5086):     x12 0000000000000003  x13 00000000ffffffff  x14 0000007fd73fa6a4  x15 0000007fd73fa23c
01-24 15:28:13.646 F/DEBUG   ( 5086):     x16 0000007c00ef11b0  x17 0000007c00e87120  x18 0000007fd73fa23a  x19 0000007c0082d2c0
01-24 15:28:13.646 F/DEBUG   ( 5086):     x20 0000005dc122ebf8  x21 0000007c0082b0b0  x22 0000007c0082b0a0  x23 0000005dc11cf6ff
01-24 15:28:13.647 F/DEBUG   ( 5086):     x24 0000005dc11dbf2a  x25 0000005dc11cf78e  x26 000000000000028d  x27 0000007c00831480
01-24 15:28:13.647 F/DEBUG   ( 5086):     x28 0000007c008314a8  x29 0000000000000018
01-24 15:28:13.647 F/DEBUG   ( 5086):     sp  0000007fd73fab90  lr  0000005dc105c084  pc  0000005dc105c090
01-24 15:28:13.648 I/unwind  ( 5086): Malformed section header found, ignoring...
01-24 15:28:13.651 F/DEBUG   ( 5086): 
01-24 15:28:13.651 F/DEBUG   ( 5086): backtrace:
01-24 15:28:13.651 F/DEBUG   ( 5086):     #00 pc 0000000000017090  /data/local/tmp/androidnn/label_image
01-24 15:28:13.651 F/DEBUG   ( 5086):     #01 pc 00000000000179e0  /data/local/tmp/androidnn/label_image
01-24 15:28:13.651 F/DEBUG   ( 5086):     #02 pc 0000000000017a88  /data/local/tmp/androidnn/label_image
01-24 15:28:13.652 F/DEBUG   ( 5086):     #03 pc 00000000000aeeb4  /system/lib64/libc.so (__libc_init+88)
01-24 15:28:14.039 I/crash_dump64( 5086): Start dumpstate
01-24 15:28:14.054 W/NativeCrashListener( 5488): Couldn't find ProcessRecord for pid 4787
01-24 15:28:14.057 E//system/bin/tombstoned( 5616): Tombstone written to: /data/tombstones/tombstone_01
01-24 15:28:14.062 E/audit   ( 5261): type=1701 audit(1548323894.051:192): auid=4294967295 uid=0 gid=0 ses=4294967295 subj=u:r:su:s0 pid=4787 comm="label_image" exe="/data/local/tmp/androidnn/label_image" sig=11 res=1
01-24 15:28:14.071 I/BootReceiver( 5488): Copying /data/tombstones/tombstone_01 to DropBox (SYSTEM_TOMBSTONE)
01-24 15:28:14.117 I/HqmInfo::LogAnalyzer( 5488): ContextBroadcastReceiver : received DropBoxManager "SYSTEM_TOMBSTONE" event
01-24 15:28:14.118 I/HqmInfo::LogAnalyzer( 5488): MSG_TYPE:MSG_APP_CRASH_CHECK_REQ
01-24 15:28:14.120 W/BroadcastQueue( 5488): Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
01-24 15:28:14.121 I/HqmInfo::AppCrashAnalyzer( 5488): checkAppError: list is null
01-24 15:28:14.123 D/dumpstate( 5093): Loading stats from /data/log/dumpstate-stats.txt
01-24 15:28:14.124 I/dumpstate( 5093): Average max progress: 1492 in 1 runs; estimated max: 1492
01-24 15:28:14.132 I/dumpstate( 5093): begin
01-24 15:28:14.133 I/dumpstate( 5093): dumpstate info: id=2, args='/system/bin/dumpstate -k -z', extra_options= app_native)
01-24 15:28:14.133 I/dumpstate( 5093): bugreport format version: 2.0
01-24 15:28:14.134 D/dumpstate( 5093): Bugreport dir: /data/log
01-24 15:28:14.134 D/dumpstate( 5093): Base name: dumpstate_app_native
01-24 15:28:14.134 D/dumpstate( 5093): Suffix: 2019-01-24-15-28-14
01-24 15:28:14.134 D/dumpstate( 5093): Log path: /data/log/dumpstate_app_native-2019-01-24-15-28-14-dumpstate_log-5093.txt
01-24 15:28:14.134 D/dumpstate( 5093): Temporary path: /data/log/dumpstate_app_native-2019-01-24-15-28-14.tmp
01-24 15:28:14.134 D/dumpstate( 5093): Screenshot path: 
01-24 15:28:14.134 D/dumpstate( 5093): Creating initial .zip file (/data/log/dumpstate_app_native-2019-01-24-15-28-14.zip)
01-24 15:28:14.135 D/dumpstate( 5093): Adding zip text entry version.txt
01-24 15:28:14.136 I/dumpstate( 5093): Vibrate: 'cmd vibrator vibrate 150 dumpstate'
01-24 15:28:14.143 I/ActivityManager( 5488): Killing 3567:com.samsung.android.bixby.agent/5018 (adj 906): empty #31
01-24 15:28:14.146 W/libprocessgroup( 5488): kill(-3567, 9) failed: No such process
01-24 15:28:14.154 W/BroadcastQueue( 5488): Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
01-24 15:28:14.196 I/chatty  ( 5488): uid=1000(system) ActivityManager identical 5 lines
01-24 15:28:14.202 W/libprocessgroup( 5488): kill(-3567, 9) failed: No such process
01-24 15:28:14.206 V/VibratorService( 5488): vibrate - package: dumpstate, token: com.android.server.VibratorService@62e6ba4, usage: 0, effect: OneShot{mDuration=150, mAmplitude=-1, mMagnitude=-1, mMagnitudeType=TYPE_EXTRA}, mag: 10000, TYPE_EXTRA
01-24 15:28:14.206 D/VibratorService( 5488): Turning vibrator off
01-24 15:28:14.207 D/SecVibrator-HAL( 5296): writeNode node:/sys/class/timed_output/vibrator/enable val:0
01-24 15:28:14.211 W/libprocessgroup( 5488): kill(-3567, 9) failed: No such process
01-24 15:28:14.213 D/VibratorService( 5488): vibratorOn() : 150ms, amplitude :-1, mag :10000, f : 0
01-24 15:28:14.215 D/SecVibrator-HAL( 5296): writeNode node:/sys/class/timed_output/vibrator/multi_freq val:0
01-24 15:28:14.217 D/SecVibrator-HAL( 5296): writeNode node:/sys/class/timed_output/vibrator/intensity val:10000
01-24 15:28:14.217 D/SecVibrator-HAL( 5296): writeNode node:/sys/class/timed_output/vibrator/enable val:150
01-24 15:28:14.219 W/libprocessgroup( 5488): kill(-3567, 9) failed: No such process
01-24 15:28:14.293 I/chatty  ( 5488): uid=1000(system) ActivityManager identical 9 lines
01-24 15:28:14.301 W/libprocessgroup( 5488): kill(-3567, 9) failed: No such process
01-24 15:28:14.305 V/[DMS]SemDesktopModeStateNotifier( 5488): binderDied(): DesktopModeListenerInfo(name=com.samsung.android.bixby.agent.app.BixbyApplication$$Lambda$1@bfd57fd, pid=3567, uid=5018)
01-24 15:28:14.307 W/libprocessgroup( 5488): kill(-3567, 9) failed: No such process
01-24 15:28:14.309 D/ForegroundUtils(21076): Process died; UID 5018 PID 3567
01-24 15:28:14.309 D/ForegroundUtils(21076): could not check pending caller
01-24 15:28:14.309 D/ForegroundUtils(21076): Foreground changed, PID: 3567 UID: 5018 foreground: false
01-24 15:28:14.309 D/ForegroundUtils(21076): Foreground UID/PID combinations:
01-24 15:28:14.310 D/ForegroundUtils(21076): UID: 10110 PID: 23150
01-24 15:28:14.323 W/libprocessgroup( 5488): kill(-3567, 9) failed: No such process
01-24 15:28:14.323 I/libprocessgroup( 5488): Successfully killed process cgroup uid 5018 pid 3567 in 177ms
01-24 15:28:14.328 I/Zygote  ( 5267): Process 3567 exited due to signal (9)
01-24 15:28:14.369 D/VibratorService( 5488): Turning vibrator off
01-24 15:28:14.370 D/SecVibrator-HAL( 5296): writeNode node:/sys/class/timed_output/vibrator/enable val:0
01-24 15:28:14.376 D/[a11y]  ( 5488): getUserAccounts0
01-24 15:28:14.379 I/chatty  ( 5488): uid=1000(system) Binder:5488_11 identical 1 line
01-24 15:28:14.381 D/[a11y]  ( 5488): getUserAccounts0
01-24 15:28:14.401 E/dumpstate( 5093): can't find the pid
01-24 15:28:14.402 E/dumpstate( 5093): Failed to find: /data/misc/anrd/
01-24 15:28:14.402 D/dumpstate( 5093): Skipping systrace because '/sys/kernel/debug/tracing/tracing_on' content is '0'
01-24 15:28:14.402 D/dumpstate( 5093): /data/misc/raft does not exist or is not a directory
01-24 15:28:14.403 D/dumpstate( 5093): Adding dir /cache/recovery (recursive: 1)
01-24 15:28:14.406 D/[a11y]  ( 5488): getUserAccounts0
01-24 15:28:14.649 D/dumpstate( 5093): Duration of '/cache/recovery': 0.246s
01-24 15:28:14.649 D/dumpstate( 5093): Adding dir /data/misc/recovery (recursive: 1)
01-24 15:28:14.650 D/dumpstate( 5093): Duration of '/data/misc/recovery': 0.001s
01-24 15:28:14.650 D/dumpstate( 5093): Adding dir /data/misc/update_engine_log (recursive: 1)
01-24 15:28:14.650 D/dumpstate( 5093): Duration of '/data/misc/update_engine_log': 0.000s
01-24 15:28:14.650 D/dumpstate( 5093): Adding dir /data/misc/logd (recursive: 0)
01-24 15:28:14.651 D/dumpstate( 5093): Duration of '/data/misc/logd': 0.000s
01-24 15:28:14.651 D/dumpstate( 5093): Adding dir /data/misc/profiles/cur (recursive: 1)
01-24 15:28:14.765 D/dumpstate( 5093): Duration of '/data/misc/profiles/cur': 0.114s
01-24 15:28:14.765 D/dumpstate( 5093): Adding dir /data/misc/profiles/ref (recursive: 1)
01-24 15:28:14.815 D/dumpstate( 5093): Duration of '/data/misc/profiles/ref': 0.050s
01-24 15:28:14.858 D/dumpstate( 5093): Adding zip text entry systemserver_fileinfo.txt
01-24 15:28:14.858 D/dumpstate( 5093): Duration of 'FILE INFO': 0.043s  ```

The default AI Benchmark successfully runs all the models which definitely has softmax operation also.
Is there any known reason for softmax failure via lable_image.

**Comment by freedomtan** [comment]
It seems it's caused by NNAPI stuff not TFLite.

**Comment by shashishekhar** [comment]
@praveen-dn : @freedomtan is correct.
@praveen-dn : Which label example were you testing and how did you enable NNAPI?

**Comment by shashishekhar** [comment]
@praveen-dn: Please reopen if you are still seeing the issue.
